libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -335,14 +366,38 @@ bool AP_ICEngine::throttle_override(uint8_t &percentage)
     if (state == ICE_RUNNING &&
         idle_percent > 0 &&
         idle_percent < 100 &&
-        (int16_t)idle_percent > SRV_Channels::get_output_scaled(SRV_Channel::k_throttle))
+        idle_percent > percentage)
     {
-        percentage = (uint8_t)idle_percent;
+        percentage = idle_percent;
         return true;
     }
 
     if (state == ICE_STARTING || state == ICE_START_DELAY) {
-        percentage = (uint8_t)start_percent.get();
+        percentage = start_percent.get();
+        return true;
+    }
+
+    if (redline.flag && !(options & uint16_t(Options::DISABLE_REDLINE_GOVERNOR))) {
```

Comment:
Why the options bit? Setting the redline rpm to zero already disables this. If you insist on using the bit, why not have it restrict setting the flag?

Reply:
That was something that was discussed in a dev call a while back. Warn from the GCS that the RPM is too high, but don't take action. The options bit controls the taking action part.